File: functions/javascript/hello-world.js
/**
 * Sample AWS Lambda-style JavaScript function
 * 
 * This function takes an input with a name and returns a greeting
 * @param {Object} event - Input event from the caller
 * @param {Object} context - Execution context
 * @param {Function} callback - Callback function to return results
 * @returns {Object} - Response object with greeting
 */
exports.handler = (event, context, callback) => {
  console.log('Function execution started');
  console.log('Input event:', JSON.stringify(event));
  
  try {
    const name = event.name || 'World';
    const response = {
      statusCode: 200,
      body: {
        message: `Hello, ${name}!`,
        timestamp: new Date().toISOString()
      }
    };
    
    console.log('Function execution completed successfully');
    
    // Return the response either via callback or direct return (supports both)
    if (callback) {
      callback(null, response);
    }
    return response;
  } catch (error) {
    console.error('Function execution failed:', error);
    
    const errorResponse = {
      statusCode: 500,
      body: {
        message: 'Error processing request',
        error: error.message
      }
    };
    
    if (callback) {
      callback(error);
    }
    throw error;
  }
};

File: functions/python/hello_world.py
"""
Sample AWS Lambda-style Python function

This function takes an input with a name and returns a greeting
"""
import json
import datetime

def handler(event, context):
    """
    Lambda handler function
    
    Args:
        event: Input event from the caller
        context: Execution context
    
    Returns:
        dict: Response object with greeting
    """
    print("Function execution started")
    print(f"Input event: {json.dumps(event)}")
    
    try:
        name = event.get('name', 'World')
        response = {
            'statusCode': 200,
            'body': {
                'message': f"Hello, {name}!",
                'timestamp': datetime.datetime.now().isoformat()
            }
        }
        
        print("Function execution completed successfully")
        return response
    except Exception as e:
        print(f"Function execution failed: {str(e)}")
        
        error_response = {
            'statusCode': 500,
            'body': {
                'message': 'Error processing request',
                'error': str(e)
            }
        }
        
        raise Exception(str(e))